Cannot compile my project with AGP >=8.0.0 due to R8 error

## Steps to reproduce
1. Create a new Flutter project.
2. Add `flutter_secure_storage: ^9.2.2` to `dependencies` of `pubspec.yaml`
3. Change AGP version to 8.5.0 in `android/settings.gradle` (`com.android.application` plugin) or any other 8.x.y.
4. Change Gradle Wrapper version to 8.7 in `android/gradle/wrapper/gradle-wrapper.properties` or any other which support the above specified AGP version.
5. Build or run Flutter app in release mode (`flutter build apk` or `flutter run --release`).

## Expected behavior
Builds successfully

## Actual behavior
```
ERROR: Missing classes detected while running R8. Please add the missing classes or apply additional keep rules that are generated in /home/evgfilim1/Projects/flutter_experiments/build/app/outputs/mapping/devRelease/missing_rules.txt.
ERROR: R8: Missing class com.google.errorprone.annotations.CanIgnoreReturnValue (referenced from: com.google.crypto.tink.KeysetManager com.google.crypto.tink.KeysetManager.add(com.google.crypto.tink.KeyTemplate) and 52 other contexts)
Missing class com.google.errorprone.annotations.CheckReturnValue (referenced from: com.google.crypto.tink.InsecureSecretKeyAccess and 1 other context)
Missing class com.google.errorprone.annotations.Immutable (referenced from: com.google.crypto.tink.InsecureSecretKeyAccess and 40 other contexts)
Missing class com.google.errorprone.annotations.RestrictedApi (referenced from: com.google.crypto.tink.aead.AesEaxKey$Builder com.google.crypto.tink.aead.AesEaxKey.builder() and 6 other contexts)
Missing class javax.annotation.Nullable (referenced from: java.lang.Object com.google.crypto.tink.PrimitiveSet$Entry.fullPrimitive and 86 other contexts)
Missing class javax.annotation.concurrent.GuardedBy (referenced from: com.google.crypto.tink.proto.Keyset$Builder com.google.crypto.tink.KeysetManager.keysetBuilder and 3 other contexts)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:minifyDevReleaseWithR8'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.R8Task$R8Runnable
   > Compilation failed to complete

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 35s
```

## Versions
- Flutter (Channel beta, 3.23.0-0.1.pre, on Arch Linux 6.9.7-zen1-1-zen, locale en_US.UTF-8)
- Android SDK version 35.0.0
- flutter_secure_storage: ^9.2.2

## Additional notes
- The issue is not reproducible on AGP <8.0.0
- Adding `android.enableR8.fullMode=false` to `android/gradle.properties` doesn't help

ERROR: Missing classes detected while running R8. Please add the missing classes or apply additional keep rules that are generated in /Users/macbook/projects/tenh100-ecom/build/app/outputs/mapping/release/missing_rules.txt.
ERROR: R8: Missing class org.slf4j.impl.StaticLoggerBinder (referenced from: void org.slf4j.LoggerFactory.bind() and 3 other contexts)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:minifyReleaseWithR8'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.R8Task$R8Runnable
   > Compilation failed to complete

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

--- a/flutter_secure_storage/android/build.gradle
+++ b/flutter_secure_storage/android/build.gradle
@@ -44,5 +44,8 @@ android {
 }
 
 dependencies {
-    implementation "androidx.security:security-crypto:1.1.0-alpha06"
+    implementation 'androidx.security:security-crypto:1.1.0-alpha06'
+
+    // Force the specific version of Tink
+    implementation 'com.google.crypto.tink:tink-android:1.9.0'
 }

